File: test/fixtures/csharp-duplicates/Readiness.cs
using System.Collections.Generic;

namespace Scheduling;

public sealed record Step(string Status, IReadOnlyList<Step> Dependencies, string? Assignee);

public static class Readiness
{
    public static bool CanStart(Step step)
    {
        int finished = 0;
        foreach (Step dependency in step.Dependencies)
            if (dependency.Status == "done") finished++;
        return step.Status == "todo" && finished == step.Dependencies.Count && step.Assignee != null;
    }
}

File: test/fixtures/rust-duplicates/src/readiness.rs
use crate::Task;

pub fn can_start(task: &Task) -> bool {
    let open_blockers = task.blockers.iter().filter(|blocker| blocker.status != "done").count();
    task.status == "todo" && open_blockers == 0 && task.assignee.is_some()
}

pub fn is_done(task: &Task) -> bool {
    task.status == "done"
}
